File: Recursion and Backtracking/Recursion with arrayList/GetKpc.cpp
#include<bits/stdc++.h>
using namespace std;

vector<string> getKpc(string str)
{
    if(str.length()==0)
    {
        vector<string> a;
        a.push_back("");
        return a;
    }
    
    string codes[] = {".;","abc","def","ghi","jkl","mno","pqrs","tu","vwx","yz"};
    char ch=str[0];
    string ros=str.substr(1);
    vector<string> recur=getKpc(ros);
    
    vector<string> ans;
    string code=codes[ch-'0'];
    for(int i=0;i<code.length();i++)
    {
        for(int j=0;j<recur.size();j++)
        {
            ans.push_back(code[i]+recur[j]);
        }
    }
    
    return ans;
}

int main()
{
    string s;
    cin >> s;
    vector<string> ans = getKpc(s);
    int cnt = 0;
    
    cout << '[';
    for (string str : ans)
    {
        if (cnt != ans.size() - 1)
            cout << str << ", ";
        else
            cout << str;
        cnt++;
    }
    cout << ']';
}

File: Recursion and Backtracking/Recursion with arrayList/GetKpc.java
import java.io.*;
import java.util.*;

public class GetKpc
{
    static String codes[] = {".;","abc","def","ghi","jkl","mno","pqrs","tu","vwx","yz"};

    public static void main(String[] args) throws Exception
    {
            Scanner sc = new Scanner(System.in);
            String  s= sc.next();
            ArrayList<String> ans = getKPC(s);
            System.out.println(ans);
            
    } 
    public static ArrayList<String> getKPC(String str)
    {
        if(str.length()==0)
        {
            ArrayList<String> a=new ArrayList<>();
            a.add("");
            return a;
        }

        char ch=str.charAt(0);
        String ros=str.substring(1);
        ArrayList<String> recur=getKPC(ros); //faith

        ArrayList<String> ans=new ArrayList<>();

        String code=codes[ch-'0'];
        for(int i=0;i<code.length();i++)
        {
            for(int j=0;j<recur.size();j++)
            {
                ans.add(code.charAt(i)+recur.get(j));
            }
        }
        return ans;
    }

}
